ddtrace/tracer/data_streams.go

```diff
@@ -0,0 +1,58 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package tracer
+
+import (
+	"context"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/datastreams"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/internal"
+)
+
+// GetDataStreamsProcessor returns the processor tracking data streams stats
+func (t *tracer) GetDataStreamsProcessor() *datastreams.Processor {
+	return t.dataStreams
+}
+
+// SetDataStreamsCheckpoint sets a consume or produce checkpoint in a Data Streams pathway.
+// This enables tracking data flow & end to end latency.
+// To learn more about the data streams product, see: https://docs.datadoghq.com/data_streams/go/
+func SetDataStreamsCheckpoint(ctx context.Context, edgeTags ...string) (p datastreams.Pathway, outCtx context.Context, ok bool) {
+	return SetDataStreamsCheckpointWithParams(ctx, datastreams.NewCheckpointParams(), edgeTags...)
+}
+
+// SetDataStreamsCheckpointWithParams sets a consume or produce checkpoint in a Data Streams pathway.
+// This enables tracking data flow & end to end latency.
+// To learn more about the data streams product, see: https://docs.datadoghq.com/data_streams/go/
+func SetDataStreamsCheckpointWithParams(ctx context.Context, params datastreams.CheckpointParams, edgeTags ...string) (p datastreams.Pathway, outCtx context.Context, ok bool) {
+	if t, ok := internal.GetGlobalTracer().(datastreams.ProcessorContainer); ok {
+		if processor := t.GetDataStreamsProcessor(); processor != nil {
+			p, outCtx = processor.SetCheckpointWithParams(ctx, params, edgeTags...)
+			return p, outCtx, true
+		}
+	}
+	return datastreams.Pathway{}, ctx, false
+}
+
+// TrackKafkaCommitOffset should be used in the consumer, to track when it acks offset.
+// if used together with TrackKafkaProduceOffset it can generate a Kafka lag in seconds metric.
+func TrackKafkaCommitOffset(group, topic string, partition int32, offset int64) {
+	if t, ok := internal.GetGlobalTracer().(datastreams.ProcessorContainer); ok {
+		if p := t.GetDataStreamsProcessor(); p != nil {
+			p.TrackKafkaCommitOffset(group, topic, partition, offset)
+		}
+	}
+}
+
+// TrackKafkaProduceOffset should be used in the producer, to track when it produces a message.
+// if used together with TrackKafkaCommitOffset it can generate a Kafka lag in seconds metric.
+func TrackKafkaProduceOffset(topic string, partition int32, offset int64) {
```

Comment:
This looks fine to me, but to be clear, this is becoming a public API, even though we are really only using it within the module right now, and we don't really expect users to directly use it.

The consequence of this is that this API cannot change at all once we release it.
If this is not a stable, complete API, and is still being developed at all, it may be better to find some way to hide it so it's not exposed publicly.

Reply:
Yeah. given that not everyone uses the sarama or confluent clients, and not everyone stores offsets in Kafka, I think it's important to leave that options to customers.
The main change I see for this API, is an additional argument (Kafka cluster) that we might add in the future.
Maybe I can add it directly, but ignore it for now?

Reply:
actually, I don't know what would be the API today. maybe a list of strings, maybe just one string.
So I think what we can do is add another function, but leave this one once we know how we will track clusters.